hazelcast/src/main/java/com/hazelcast/nio/serialization/PortableReader.java

```diff
@@ -36,165 +43,285 @@
      */
-    int readInt(String fieldName) throws IOException;
+    int readInt(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the long value read
      * @throws IOException in case of any exceptional case
      */
-    long readLong(String fieldName) throws IOException;
+    long readLong(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the utf string value read
      * @throws IOException in case of any exceptional case
      */
-    String readUTF(String fieldName) throws IOException;
+    @Nullable
+    String readUTF(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the boolean value read
      * @throws IOException in case of any exceptional case
      */
-    boolean readBoolean(String fieldName) throws IOException;
+    boolean readBoolean(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the byte value read
      * @throws IOException in case of any exceptional case
      */
-    byte readByte(String fieldName) throws IOException;
+    byte readByte(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the char value read
      * @throws IOException in case of any exceptional case
      */
-    char readChar(String fieldName) throws IOException;
+    char readChar(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the double value read
      * @throws IOException in case of any exceptional case
      */
-    double readDouble(String fieldName) throws IOException;
+    double readDouble(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the float value read
      * @throws IOException in case of any exceptional case
      */
-    float readFloat(String fieldName) throws IOException;
+    float readFloat(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @return the short value read
      * @throws IOException in case of any exceptional case
      */
-    short readShort(String fieldName) throws IOException;
+    short readShort(@Nonnull String fieldName) throws IOException;
 
     /**
      * @param fieldName name of the field
      * @param <P>       the type of the portable read
      * @return the portable value read
      * @throws IOException in case of any exceptional case
      */
-    <P extends Portable> P readPortable(String fieldName) throws IOException;
+    @Nullable
+    <P extends Portable> P readPortable(@Nonnull String fieldName) throws IOException;
+
+    /**
+     * Reads a decimal which is arbitrary precision and scale floating-point number to BigDecimal
```

Comment:
nit: missing trailing dot here and in below
```suggestion
     * Reads a decimal which is arbitrary precision and scale floating-point number to BigDecimal.
```